File: backend/models/filters.py
from enum import Enum


class WorkMode(str, Enum):
    ONSITE = "onsite"
    REMOTE = "remote"
    HYBRID = "hybrid"


class JobType(str, Enum):
    FULL_TIME = "full_time"
    PART_TIME = "part_time"
    CONTRACT = "contract"
    TEMPORARY = "temporary"
    INTERNSHIP = "internship"
    VOLUNTEER = "volunteer"
    FREELANCE = "freelance"


class ExperienceLevel(str, Enum):
    INTERNSHIP = "internship"
    ENTRY = "entry"
    ASSOCIATE = "associate"
    MID_SENIOR = "mid_senior"
    DIRECTOR = "director"
    EXECUTIVE = "executive"


class DatePosted(str, Enum):
    ANY_TIME = "any_time"
    PAST_MONTH = "past_month"
    PAST_WEEK = "past_week"
    PAST_24H = "past_24h"


WORK_MODE_LABELS = {
    WorkMode.ONSITE: "On-site",
    WorkMode.REMOTE: "Remote",
    WorkMode.HYBRID: "Hybrid",
}

JOB_TYPE_LABELS = {
    JobType.FULL_TIME: "Full-time",
    JobType.PART_TIME: "Part-time",
    JobType.CONTRACT: "Contract",
    JobType.TEMPORARY: "Temporary",
    JobType.INTERNSHIP: "Internship",
    JobType.VOLUNTEER: "Volunteer",
    JobType.FREELANCE: "Freelance",
}

EXPERIENCE_LEVEL_LABELS = {
    ExperienceLevel.INTERNSHIP: "Internship",
    ExperienceLevel.ENTRY: "Entry level",
    ExperienceLevel.ASSOCIATE: "Associate",
    ExperienceLevel.MID_SENIOR: "Mid-Senior level",
    ExperienceLevel.DIRECTOR: "Director",
    ExperienceLevel.EXECUTIVE: "Executive",
}

DATE_POSTED_LABELS = {
    DatePosted.ANY_TIME: "Any time",
    DatePosted.PAST_MONTH: "Past month",
    DatePosted.PAST_WEEK: "Past week",
    DatePosted.PAST_24H: "Past 24 hours",
}

File: frontend/lib/filters.ts
export const WORK_MODE_OPTIONS = [
  { value: "onsite", label: "On-site" },
  { value: "remote", label: "Remote" },
  { value: "hybrid", label: "Hybrid" },
] as const

export const JOB_TYPE_OPTIONS = [
  { value: "full_time", label: "Full-time" },
  { value: "part_time", label: "Part-time" },
  { value: "contract", label: "Contract" },
  { value: "temporary", label: "Temporary" },
  { value: "internship", label: "Internship" },
  { value: "volunteer", label: "Volunteer" },
  { value: "freelance", label: "Freelance" },
] as const

export const EXPERIENCE_LEVEL_OPTIONS = [
  { value: "internship", label: "Internship" },
  { value: "entry", label: "Entry level" },
  { value: "associate", label: "Associate" },
  { value: "mid_senior", label: "Mid-Senior level" },
  { value: "director", label: "Director" },
  { value: "executive", label: "Executive" },
] as const

export const DATE_POSTED_OPTIONS = [
  { value: "any_time", label: "Any time" },
  { value: "past_month", label: "Past month" },
  { value: "past_week", label: "Past week" },
  { value: "past_24h", label: "Past 24 hours" },
] as const

export type WorkMode = typeof WORK_MODE_OPTIONS[number]["value"]
export type JobType = typeof JOB_TYPE_OPTIONS[number]["value"]
export type ExperienceLevel = typeof EXPERIENCE_LEVEL_OPTIONS[number]["value"]
export type DatePosted = typeof DATE_POSTED_OPTIONS[number]["value"]

export interface JobFilters {
  query: string
  country: string
  city: string
  work_mode: WorkMode[]
  job_type: JobType[]
  experience_level: ExperienceLevel[]
  date_posted: DatePosted | ""
  salary_min: string
  salary_max: string
}

export const defaultFilters: JobFilters = {
  query: "",
  country: "",
  city: "",
  work_mode: [],
  job_type: [],
  experience_level: [],
  date_posted: "",
  salary_min: "",
  salary_max: "",
}
